helpers/functions.php

```diff
@@ -7,19 +7,24 @@
 use LaravelLang\Config\Facades\Config;
 
 if (! function_exists('localizedRoute')) {
-    function localizedRoute(string $route, array $parameters = [], bool $absolute = true): string
-    {
-        $locale = Config::shared()->routes->names->parameter;
-        $prefix = Config::shared()->routes->namePrefix;
+    function localizedRoute(
+        string $route,
+        array $parameters = [],
+        bool $absolute = true,
+        bool $redirectDefault = false
+    ): string {
+        $currentLocale = app()->getLocale();
+        $defaultLocale = config('app.locale');
 
-        $name = Str::start($route, $prefix);
-
-        if (! Route::has($name)) {
+        if ($redirectDefault && $currentLocale === $defaultLocale) {
             return route($route, $parameters, $absolute);
         }
 
-        return route($name, array_merge([
-            $locale => app()->getLocale(),
-        ], $parameters), $absolute);
+        $config = Config::shared()->routes;
+        $name = Str::start($route, $config->namePrefix);
+
+        return Route::has($name)
+            ? route($name, [$config->names->parameter => $currentLocale] + $parameters, $absolute)
```

Comment:
You need to use `array_merge` because explicitly declared keys must be overwritten.